Nightly build error, intel/2023.1+mkl builds

Nightly testing with intel/2023.1+mkl had compilation errors of the form:

```
04:18:07 /home/jenkins/blake-new/workspace/KokkosKernels_Nightly_Blake_OneAPI_2023_1_0_OpenMP_Serial_SPR-oneMKL/kokkos-kernels/sparse/tpls/KokkosSparse_spmv_bsrmatrix_tpl_spec_decl.hpp:51:31: error: use of undeclared identifier 'exec'
04:18:07     subhandle->set_exec_space(exec);
04:18:07                               ^
04:18:07 /home/jenkins/blake-new/workspace/KokkosKernels_Nightly_Blake_OneAPI_2023_1_0_OpenMP_Serial_SPR-oneMKL/kokkos-kernels/sparse/tpls/KokkosSparse_spmv_bsrmatrix_tpl_spec_decl.hpp:131:31: error: use of undeclared identifier 'exec'
04:18:07     subhandle->set_exec_space(exec);
04:18:07   
```

Looks like this follows merge of #2164 , @brian-kelley can you help advise? I don't see an execution space instance passed through the `spmv_bsr_mkl` function signature, should the `exec` arguments be replaced with a default instance e.g. `ExecSpace()`?

**Reproducer** (Blake all queue):
```bash
source /projects/x86-64-icelake-rocky8/spack-config/blake-setup-user-module-env.sh
module purge
module load git cmake intel-oneapi-compilers/2023.1.0 intel-oneapi-dpl/2022.1.0 intel-oneapi-mkl/2023.1.0 intel-oneapi-tbb/2021.9.0

$KOKKOSKERNELS_PATH/cm_generate_makefile.bash --with-openmp --with-serial --arch=SPR --compiler=icpx --cxxflags="-fp-model=precise" --with-tpls=mkl --kokkos-cmake-flags=-DKokkos_ENABLE_ONEDPL=OFF -kokkos-path=$KOKKOS_PATH
```

--- a/sparse/tpls/KokkosSparse_spmv_bsrmatrix_tpl_spec_decl.hpp
+++ b/sparse/tpls/KokkosSparse_spmv_bsrmatrix_tpl_spec_decl.hpp
@@ -48,7 +48,8 @@ inline void spmv_bsr_mkl(Handle* handle, sparse_operation_t op, Scalar alpha,
     if (!subhandle)
       throw std::runtime_error(
           "KokkosSparse::spmv: subhandle is not set up for MKL BSR");
-    subhandle->set_exec_space(exec);
+    // note: classic mkl only runs on synchronous host exec spaces, so no need
+    // to call set_exec_space on the subhandle here
   } else {
     // Use the default execution space instance, as classic MKL does not use
     // a specific instance.
@@ -128,7 +129,8 @@ inline void spmv_mv_bsr_mkl(Handle* handle, sparse_operation_t op, Scalar alpha,
     if (!subhandle)
       throw std::runtime_error(
           "KokkosSparse::spmv: subhandle is not set up for MKL BSR");
-    subhandle->set_exec_space(exec);
+    // note: classic mkl only runs on synchronous host exec spaces, so no need
+    // to call set_exec_space on the subhandle here
   } else {
     // Use the default execution space instance, as classic MKL does not use
     // a specific instance.

--- a/sparse/tpls/KokkosSparse_spmv_tpl_spec_decl.hpp
+++ b/sparse/tpls/KokkosSparse_spmv_tpl_spec_decl.hpp
@@ -553,11 +553,11 @@ inline void spmv_mkl(Handle* handle, sparse_operation_t op, Scalar alpha,
   MKLScalar* y_mkl       = reinterpret_cast<MKLScalar*>(y);
   if (handle->is_set_up) {
     subhandle = dynamic_cast<Subhandle*>(handle->tpl);
-    // note: classic mkl only runs on synchronous host exec spaces, so no need
-    // to call set_exec_space on the subhandle here
     if (!subhandle)
       throw std::runtime_error(
           "KokkosSparse::spmv: subhandle is not set up for MKL CRS");
+    // note: classic mkl only runs on synchronous host exec spaces, so no need
+    // to call set_exec_space on the subhandle here
   } else {
     // Use the default execution space instance, as classic MKL does not use
     // a specific instance.

